### ISSUE TITLE
🐛 Bug: Incorrect positioning when using tooltip and dropdown builders

### ISSUE THREAD

**Issue by mihar-22** [melt-ui/melt-ui#276]
👋 

Loving the library so far, thank you so much for your work. I'm having a little issue when trying to compose multiple builders, in this case the tooltip and dropdown menu. It seems to position the tooltip incorrectly. Am I doing something wrong is this a possible bug?

<details>
  <summary>Code</summary>
  <img width="906" alt="image" src="https://github.com/melt-ui/melt-ui/assets/14304599/f2755307-4be5-4308-8ddb-3301f899b5f0">
</details>

<details>
  <summary>Result</summary>
  
  Hovering over the screen icon on the far right produces this tooltip positioned on the far left.
<img width="811" alt="image" src="https://github.com/melt-ui/melt-ui/assets/14304599/083fbfb6-2184-4057-9e9e-8b5f42fdde8c">
</details>

**Comment by huntabyte** [comment]
Can you drop a Stackblitz repro link so I can further investigate? I haven't actually tried combining them like that before so I'm intrigued!

**Comment by mihar-22** [comment]
Sure!

https://stackblitz.com/edit/melt-ui-playground-raaejq?file=src%2Froutes%2F%2Bpage.svelte

**Comment by huntabyte** [comment]
Yeah, I definitely see the issue, and I think this is likely a common use case (hover over the button to see what it does and click to open a dropdown). 

I'll work on pinpointing this issue, if it's not possible with the current APIs, we may be able to make it possible with our "Big Bad" refactor we're working on.

**Comment by huntabyte** [comment]
Okay so after looking into this a bit it makes more sense. Basically what's happening is you're overriding one of the attributes of the component or the other (dropdown & tooltip). The other elements use some of those attributes to determine how they will behave, which I've come to learn may be an issue on our end.

As a temporary fix, if you spread the tooltip store last on the element, the tooltip will work as expected, but it may have unexpected effects on the dropdown menu, the example repro doesn't have a menu buildout for me to see that for certain, but I'd appreciate it if you could let us know.

I'm glad you opened this issue because it has caused me to find some issues in our tooltip implementation.

More specifically, both `triggers` (tooltip and dropdown) apply `aria-controls` an attribute, which I'm not sure is necessary for the tooltip according to this [pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/) (but it's late and I could be missing something).

So we may need to make some changes there. Additionally, our tooltip doesn't apply the `aria-describedby` attribute to the trigger which should reference the id of the tooltip content, so we potentially don't need an `id` on the tooltip at all, which I believe would get rid of most conflicts it may have with the dropdown.

I'll defer to @TGlide for his thoughts on the matter before taking any action though!

Thanks again for raising this issue @mihar-22!

**Comment by mihar-22** [comment]
Thanks, appreciate the update!

Adding my notes on the issues I'm seeing:

-  Setting tooltip trigger after menu trigger works, but the the tooltip remains visible when the menu is closed and the pointer is _not_ on the tooltip trigger.
- The `aria-describedby` attribute is missing on the tooltip trigger on hover/focus.
- The `role="tooltip"` is missing on tooltip content.
- The `aria-controls` should not be set for a tooltip trigger.
- The `aria-expanded` attribute should not be set for a tooltip trigger.
- The `aria-expanded` attribute not updating correctly when both a tooltip/menu trigger is applied.

**Comment by huntabyte** [comment]
Thanks @mihar-22! @TGlide & I have discussed this extensively and I have submitted a PR that _should_ fix this issue for you, but I'd appreciate a confirmation once the updates are released if it's behaving as you'd expect!

**Comment by mihar-22** [comment]
Yep can confirm all bugs have been resolved! Thank you.

There's only a slight issue if you open a menu with a tooltip, and then click away to hide the menu, the tooltip doesn't disappear (requires second click). It's not a big deal and I can open a separate issue later, main things are all fixed.
